internal/engine/buildcontrol/docker_compose_build_and_deployer.go

```diff
@@ -138,18 +141,31 @@ func (bd *DockerComposeBuildAndDeployer) BuildAndDeploy(ctx context.Context, st
 		ps.EndPipelineStep(ctx)
 	}
 
-	iTargetMap := model.ImageTargetsByID(plan.tiltManagedImageTargets)
+	var cluster v1alpha1.Cluster
+	// If the cluster fetch fails, that's OK.
```

Comment:
FWIW, I found this confusing:
* if the cluster is an input to the build, does a failure here means we're building with incorrect parameters?
* if it's not, why call `Get` rather than passing a default Cluster?

Reply:
ya, that's fair. i removed it.

in theory, you could have a DockerCompose cluster with a different arch than your host cluster, but it seems unlikely to happen in practice.

this will all go away once we move these into a reconciler, since the image builder won't know if it's building for docker compose or k8s